swap_meet/vendor.py

```diff
@@ -1,2 +1,85 @@
 class Vendor:
-    pass
+    
+    def __init__(self, inventory=[]):
+        self.inventory = inventory
+    
+    def add(self, item):
+        self.inventory.append(item)
+        return item
+    
+    def remove(self, item):
+        if item not in self.inventory:
+            return False
+        self.inventory.remove(item)
+        return item
+    
+    def get_by_category(self, category):
+        matching_items = []
+        for item in self.inventory:
+            if item.category == category:
+                matching_items.append(item)
+        return matching_items
```

Comment:
We can refactor this function using list comprehension:

` matching_items =  [
            item for item in self.inventory if item.category == category]